src/shared/directives/conditional-show/services/condition-parser.service.ts

```diff
@@ -0,0 +1,204 @@
+        (_score.has(dynamiclist, 'value') && _score.has(dynamiclist, 'list_items'));
+  }
+
+  private static getField(condition: string): [string, string?] {
+    let separator: string = ShowCondition.CONTAINS;
+    if (condition.indexOf(ShowCondition.CONTAINS) < 0) {
+      separator = ShowCondition.CONDITION_EQUALS;
+      if (condition.indexOf(ShowCondition.CONDITION_NOT_EQUALS) > -1) {
+        separator = ShowCondition.CONDITION_NOT_EQUALS;
+      }
+    }
+    return [ condition.split(separator)[0], separator ];
+  }
+
+  private static checkValueEquals(expectedValue: string, currentValue: any, conditionSeparaor: string): boolean {
+    if (expectedValue.search('[,]') > -1) { // for  multi-select list
+      return this.checkMultiSelectListEquals(expectedValue, currentValue, conditionSeparaor);
+    } else if (expectedValue.endsWith('*') && currentValue && conditionSeparaor !== ShowCondition.CONDITION_NOT_EQUALS) {
+      if (typeof currentValue === 'string') {
+        return currentValue.startsWith(this.removeStarChar(expectedValue));
+      }
+      return expectedValue === '*';
+    } else {
+      // changed from '===' to '==' to cover number field conditions
+      if (conditionSeparaor === ShowCondition.CONDITION_NOT_EQUALS) {
+        return this.checkValueNotEquals(expectedValue, currentValue);
+      } else {
+        return currentValue == expectedValue || this.okIfBothEmpty(expectedValue, currentValue); // tslint:disable-line
+      }
+    }
+  }
+
+  private static checkValueNotEquals(expectedValue: string, currentValue: any): boolean {
+    const formatCurrentValue = currentValue ? currentValue.toString().trim() : '';
+    if ('*' === expectedValue && formatCurrentValue !== '') {
+      return false;
+    }
+    const formatExpectedValue = expectedValue ? expectedValue.toString().trim() : '';
+    return formatCurrentValue != formatExpectedValue; // tslint:disable-line
+  }
+
+  private static checkMultiSelectListEquals(expectedValue: string, currentValue: any, conditionSeparator: string): boolean {
+    const expectedValues = expectedValue.split(',').sort().toString();
+    const values = currentValue ? currentValue.sort().toString() : '';
+    if (conditionSeparator === ShowCondition.CONDITION_NOT_EQUALS) {
+      return expectedValues !== values;
+    } else {
+      return expectedValues === values;
+    }
+  }
+
+  private static checkValueContains(expectedValue: string, currentValue: any): boolean {
+    if (expectedValue.search(',') > -1) {
+      let expectedValues = expectedValue.split(',').sort();
+      let values = currentValue ? currentValue.sort().toString() : '';
+      return expectedValues.every(item => values.search(item) >= 0);
+    } else {
+      let values = currentValue && Array.isArray(currentValue) ? currentValue.toString() : '';
+      return values.search(expectedValue) >= 0;
+    }
+  }
+
+  private static unquoted(str: string): string {
+    const res = str.replace(/^"|"$/g, '');
+    return res;
```

Comment:
const res is redudant it can be returned immediately.

Reply:
done